backend/dataall/modules/dataset_sharing/db/share_object_repositories.py

```diff
@@ -1307,3 +1307,61 @@ def count_role_principal_shares(session, principal_id: str, principal_type: Prin
             )
             .count()
         )
+
+
+class DatasetLockShareItemManagementSM:
```

Comment:
I thought it best to keep it different from ShareItemSM and ShareObjectSM since the transformations in those are very different from the ones we wanted. For example, we are going directly from Approved to Processed - which might confuse some customers.

Reply:
I think we should opt to have 1 SM manage the entire status lifecycle of the share object / share items rather than 2 separate SM classes that would manage different transitions of the same status attribute

Reply:
I think best to just add a new `ShareObjectActions` of `FailedStart` or `FailedAcquireLock` or similar name and have that start the transitions we need using the pre-existing SM classes

Reply:
Sure, I have included the change in the next revision.